synapse/storage/databases/main/group_server.py

```diff
@@ -49,13 +49,8 @@ def __init__(
         db_conn: LoggingDatabaseConnection,
         hs: "HomeServer",
     ):
-        database.updates.register_background_index_update(
-            update_name="local_group_updates_index",
-            index_name="local_group_updates_stream_id_index",
-            table="local_group_updates",
-            columns=("stream_id",),
-            unique=True,
-        )
+        # Register a legacy groups background update as a no-op.
+        database.updates.register_noop_background_update("local_group_updates_index")
```

Comment:
It would be ideal to put this on a different store, but there doesn't seem to be a good spot for it. Any thoughts?

Reply:
These no-op updates always confuse me; what would be wrong with adding `DELETE FROM background_updates WHERE name = '....';` to your schema delta?

Reply:
> These no-op updates always confuse me; what would be wrong with adding `DELETE FROM background_updates WHERE name = '....';` to your schema delta?

That could work too, I'll ask the team if that seems like it would work!

Reply:
The conclusion from richvdh was:

> I think: because I didn't think of `DELETE` when I wrote `register_noop_background_update`: 6b02fc80d173d3d4de81623d411a136abe1637e9

Reply:
I think my preference here would be to leave this as-is and handle this (and the other no-ops) in a separate PR? What do you think @reivilibre -- it is quite orthogonal to removal of groups?

Reply:
Filed #12940 about this.